Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -249,27 +249,30 @@ protected void SendRPCInternal(Type invokeClass, string rpcName, NetworkWriter w
         [EditorBrowsable(EditorBrowsableState.Never)]
         protected void SendTargetRPCInternal(NetworkConnection conn, Type invokeClass, string rpcName, NetworkWriter writer, int channelId)
         {
-            // this was in Weaver before
             if (!NetworkServer.active)
             {
-                logger.LogError("TargetRPC Function " + rpcName + " called on client.");
+                logger.LogError($"TargetRPC {rpcName} called when server not active");
                 return;
             }
-            // connection parameter is optional. assign if null.
-            if (conn == null)
+
+            if (!isServer)
             {
-                conn = connectionToClient;
+                logger.LogWarning($"TargetRpc {rpcName} called on {name} but that object has not been spawned or has been unspawned");
+                return;
             }
-            // this was in Weaver before
-            if (conn is NetworkConnectionToServer)
+
+            // connection parameter is optional. assign if null.
+            conn = conn ?? connectionToClient;
```

Comment:
nice PR, but this part was way easier to read and understand previously.
can't we keep the if == null then := connectionToClient?

Reply:
sure I can change it to the old one

Reply:
There really should be code consistency.  ?? has been used in 18 other places.  Either use it everywhere or take them out everywhere. @vis2k @James-Frowen

Reply:
`??` doesn't seem to be used must, seems like we only use it in the synclist/set constructors. Ideally used use 
`conn ??= connectionToClient;` here but that isnt supported in unity yet.